### ISSUE TITLE
Increased click latency when displaying on-screen keyboard

### ISSUE THREAD

**Issue by tsvetomir** [telerik/kendo-ui-core#1445]
There's a noticeable delay when repeatedly clicking a button if it leads to the on-screen keyboard being shown.

The issue is reproducible in the browser. A [runnable sample](http://dojo.telerik.com/@tsvetomir/IqITe) is available in the Dojo.

> If you start "spamming" the "slow" button, the device will start "queueing" up events, and this will cause a persistent latency to be added to any touch event, even after the "burst" of event ceased.
> With a few of button presses, the latency to Touch End Dispatched is less than 100ms
> With 100 button presses, the latency to Touch End Dispatched is about 300ms
> ...
>  After 300 presses, the app is basically unresponsive.
> 
> I tried the same app on an older iOS (8.1.3) and we didn't experience any event latency.
> The same app, on a iOS 9.2.1 device, experiences the increasing event latency as described before.

More details in ticket 1010562.

**Comment by Canna71** [comment]
Hi all, I would like to add that the issue happens for any event handler that takes a bit to run, not only if the keyboard is shown. For example a long enough for loop in the "slow" button, causing events to be "queued", is enough for increasing the delay in the long run.

**Comment by Canna71** [comment]
Also, by removing all js libraries and writing the same app with pure Javascript, the latency won't increase, so it seems to be some feature used by JQuery/Kendo that doesn't cooperate well with iOS browser(s).

**Comment by underlog** [comment]
@Canna71 - this seems interesting - we would definitely make use of the reference implementation you describe. May you please post it in a sample somewhere? Thanks.

**Comment by Canna71** [comment]
Here is the index.html I used in my sample Cordova app with pure javascript:
https://www.dropbox.com/s/s9gsup13brtr3fp/index.html?dl=0
And this is with jquery only:
https://www.dropbox.com/s/bma8u0yw53k5vto/index.html?dl=0
Both apps don't exhibits that strange latency degradation.
If you need, I might try adapting for Dojo or something, I still haven't tried with mobile safari.
The app in which we experienced this issue is not of course like this, (we don't ask our user to spam buttons), but there are other events going around. My suspects is that the "queueing" up of events is messing something up with some "internal timer" that causes some kind of "time dilation" in subsequent event, but only if kendo is used somewhere.

**Comment by Canna71** [comment]
Hi, upon further investigation, using the "sample app", this is something I discovered:
The focus() api seems to be determinant in causing the issue.
From the jQuery version, which is working fine, is enough to add data-role="view" and init the kendo app to have the issue.
Unfortunately, being this the case, it is hard for me to work around the issue since, even if I substitute any data and event binding with pure javascript, in my app, I would still have issues.
Note that in my real app I tried moving all of the focus() calls inside a setTimeout, but overtime the app is still getting unresponsive.

Here is the modified sample
https://www.dropbox.com/s/jor962wek6sg6nk/index.html?dl=0
